omou-frontend/src/components/BackButton.js

```diff
@@ -0,0 +1,83 @@
+import React, {Component} from "react";
+import {withRouter} from "react-router-dom";
+import PropTypes from "prop-types";
+
+import BackArrow from "@material-ui/icons/ArrowBack";
+import Modal from "@material-ui/core/Modal";
+import Button from "@material-ui/core/Button";
+import {Typography} from "@material-ui/core";
+
+class BackButton extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            warningOpen: false,
+        };
+    }
+
+    handleClick() {
+        if (this.props.warn) {
+            this.setState({
+                warningOpen: true,
+            });
+        } else {
+            this.goBack();
+        }
+    }
+
+    hideWarning() {
+        this.setState({
+            warningOpen: false,
+        });
+    }
+
+    goBack() {
+        this.props.onBack();
+        this.props.history.goBack();
+    }
+
+    render() {
+        return (
+            <div>
+                <div className="control"
+                    onClick={this.handleClick.bind(this)}>
+                    <BackArrow className="icon" />
+                    <div className="label">Back</div>
+                </div>
+                <Modal
+                    aria-labelledby="simple-modal-title"
+                    aria-describedby="simple-modal-description"
+                    open={this.state.warningOpen}
+                    onClick={this.hideWarning.bind(this)}>
+                    <div className="exit-popup">
+                        <Typography variant="h6" id="modal-title">
+                            Are you sure you want to go back?
```

Comment:
wording on this should change, we should have some custom options for what type of warnings get displayed based on props passed in. this can be used in a future feature though